### ISSUE TITLE
Remove broken `MLLP.Receiver.start/0`

### ISSUE THREAD

**Issue by bryanhunter** [HCA-Healthcare/elixir-mllp#25]
At one point `MLLP.Receiver.start/0` could be called, and it would use the DefaultDispatcher. Currently, it only returns an error.

```
iex> MLLP.Receiver.start()                                           
** (ArgumentError) No tcp port provided
    (elixir_mllp 0.5.0) lib/mllp/receiver.ex:168: MLLP.Receiver.to_args/1
    (elixir_mllp 0.5.0) lib/mllp/receiver.ex:64: MLLP.Receiver.start/1
```

**Comment by vikas15bhardwaj** [comment]
Port and dispatcher are required fields now, otherwise, it will raise error. Any idea, in which case we need to start without dispatcher.

**Comment by starbelly** [comment]
@vikas15bhardwaj The issue is that we have `start(opts \\ [])` which  defines`start/0` which will of course result  in an error. We simply need to make opts mandatory. I'll open a PR to fix this up.

**Comment by starbelly** [comment]
This was closed via #28
